testsuite/forge/src/interface/swarm.rs

```diff
@@ -76,6 +76,9 @@ pub trait Swarm: Sync {
     fn inject_chaos(&mut self, chaos: SwarmChaos) -> Result<()>;
     fn remove_chaos(&mut self, chaos: SwarmChaos) -> Result<()>;
 
+    async fn ensure_no_validator_restart(&mut self) -> Result<()>;
```

Comment:
instead of just adding a bunch of functions to the interface, how about something like `ensure_swarm_criteria(criteria)` (or similarly named), which would let us check for arbitrary (success) criteria about the state of each node/swarm?

Reply:
Yes, I was thinking of doing that when I add criteria like resource metrics. How about this - I address this in the next PR, which will add these criteria?

Reply:
sure yea sgtm